f8a_worker/workers/init_package_flow.py

```diff
@@ -115,36 +38,21 @@ def execute(self, arguments):
         except NoResultFound:
             raise FatalTaskError('Unknown ecosystem: %r' % arguments['ecosystem'])
         package = Package.get_or_create(db, ecosystem_id=ecosystem.id, name=arguments['name'])
-        url = self.get_upstream_url(arguments)
-        upstream = self.get_upstream_entry(package, url)
-        if upstream is None:
-            upstream = self.add_or_update_upstream(package, url)
-        arguments['url'] = upstream.url
+        package_analysis = db.query(PackageAnalysis) \
+            .filter(PackageAnalysis.package_id == package.id) \
+            .order_by(PackageAnalysis.started_at) \
+            .first()
 
-        if not arguments.get('force'):
-            # can potentially schedule two flows of a same type at the same
-            # time as there is no lock, but let's say it's OK
-            if upstream.updated_at is not None \
-                    and datetime.datetime.utcnow() - upstream.updated_at < self._UPDATE_INTERVAL:
-                self.log.info('Skipping upstream package check as data are considered as recent - '
-                              'last update %s.',
-                              upstream.updated_at)
-                # keep track of start, but do not schedule nothing more
-                # discard changes like updates
-                db.rollback()
+        if not arguments.get('force') and package_analysis is not None:
+            if datetime.datetime.utcnow() - package_analysis.started_at < self._UPDATE_INTERVAL:
+                InitPackageFlow.inject_arguments(arguments, package_analysis.id, package.id)
```

Comment:
Injecting `document_id` argument beats the purpose of this is-older-than condition. Presence of the argument later controls whether we should run github task and others: https://github.com/pkajaba/fabric8-analytics-worker/blob/df19a286b553b8a41460a6daf45f79d0a29a0e17/f8a_worker/dispatcher/flows/bayesianPackageFlow.yml#L23-L25. So if we care about how old the previous analysis is, then we cannot inject the `document_id` argument in this case.